imports/plugins/core/accounts/client/containers/addressBookContainer.js

```diff
@@ -0,0 +1,223 @@
+            reject(i18next.t("addressBookAdd.failedToAddAddress", { err: err.message }));
+          } else {
+            resolve(res);
+          }
+        });
+      });
+    } else {
+      Meteor.call("accounts/addressBookAdd", address, (err, res) => {
+        if (err || !res) reject(i18next.t("addressBookAdd.failedToAddAddress", { err: err.message }));
+        if (res) resolve(res);
+      });
+    }
+  });
+}
+
+/**
+ * @method onError
+ * @summary helper function that shows an error message in an alert toast.
+ * @since 2.0.0
+ * @param {Object} errorMessage - error message object.
+ */
+function onError(errorMessage) {
+  Alerts.toast(errorMessage, "error");
+}
+
+const wrapComponent = (Comp) => (
+  class AddressBookContainer extends Component {
+    static propTypes = Comp.PropTypes;
+
+    render() {
+      return (
+        <Comp
+          {...this.props}
+          addAddress={addAddress}
+          updateAddress={updateAddress}
+          removeAddress={removeAddress}
+          onError={onError}
+        />
+      );
+    }
+  }
+);
+
+function composer(props, onData) {
+  const handle = Meteor.subscribe("Accounts", Meteor.userId());
+  if (!handle.ready()) {
+    return;
+  }
+  const account = Collections.Accounts.findOne({ _id: Meteor.userId() });
+  const { addressBook } = account.profile;
+  const countries = Countries.find().fetch();
+  const shop = Collections.Shops.findOne();
+  const shopCountries = shop.locales.countries;
+
+  const regionsByCountry = {};
+  Object.keys(shopCountries).forEach((key) => {
+    const { states } = shopCountries[key] || {};
+    const regions = [];
+    if (states) {
+      // states is an object that needs to be convered
+      // to an array of region labels and values
+      Object.keys(states).forEach((i) => {
+        regions.push({
+          label: states[i].name,
+          value: i
+        });
+      });
+    }
+    regionsByCountry[key] = regions;
+  });
+  // The initial mode for addressBook
+  let initMode;
+  const cart = Collections.Cart.findOne();
+  // If we have passed the address step, show the grid
+  if (cart.workflow.status !== "checkoutAddressBook") {
+    initMode = "grid";
+  }
+  // AddressBook heading will be different in different views
+  // If the view template that's using the AddressBook has a
+  // heading object, set it as the AddressBook heading
+  if (!this.heading) {
+    const template = Template.instance();
+    const { data } = template || { data: undefined };
+    const { heading: templateHeading } = data || { heading: undefined };
+    if (templateHeading) {
+      this.heading = templateHeading;
+    } else {
+      this.heading = {
+        defaultValue: "Address Book",
+        i18nKey: "accountsUI.addressBook"
+      };
+    }
+  }
+
+  onData(null, {
+    addressBook,
+    countries,
+    heading: this.heading,
+    initMode,
+    regionsByCountry
+  });
+}
+
+registerComponent("AddressBookContainer", AddressBook, [
```

Comment:
This should be named `AddressBook` (without the `Container` in the name).  That decision was originally made because it's potentially a little confusing when you do something like...

```js
replaceComponent("AddressBookContainer", CustomAddressBook);
```
That isn't actually replacing the container.  It's replacing the component and inheriting the same container.  So anything that gets registered and added to `Components` with `registerComponent` should have the name of the UI component.